adr/RKE-KDM-decouple.md

```diff
@@ -0,0 +1,185 @@
+
+Note that, as of today, RKE1 does not intend to quickly release upstream Kubernetes releases. By the time the KDM branch is created, it is normal that upstream Kubernetes releases are already out.
+
+
+### Problems of the current release process
+
+It is impossible to have a stable cadence of releases in RKE1 because RKE1 depends on the KDM release and the KDM release cedence is unpredictable. The main reason for this unpredicatability is because Rancher Manager uses KDM for its functioning and will not allow it to be released until all tests are working. Rancher Manager is a very complex software that could require a lot of time (weeks/months) until it is ready to release.
+
+As a consequence:
+* It is impossible to know when does the releases process start, which makes it difficult to plan what gets in or not. e.g. maybe the release cycle is super short because we are already late and only a few issues can be added.
+* It is impossible to clearly state when does code freeze start or finishes
+* It is impossible for QA to plan testing ahead
+* Your PR might have been merged more than a month ago and now you get questions from QA about that PR, which is half forgotten
+* It could take months for an important update to happen in an addon (e.g. CVE)
+* Big version jumps are sometimes made, which could be risky for regressions
+* Human/conscious decisions taken, which makes automation complicated. For example, as releases can be delayed for a long time, new unplanned stuff gets sometimes sneaked into a release while waiting for KDM to officially release (e.g. CVE fix for k8s-dns)
+
+
+## Proposal
+
+Two related proposals:
+* Decouple RKE1 from Rancher Manager unpredictable release cycle as much as possible. To do so, we need to make RKE1 able to release without waiting on KDM to be officially released.
+* Do monthly releases based on the upstream Kubernetes release cadence (following rke2/k3s cadence model)
+
+My proposal is the following:
+* In KDM, continue creating a branch with the name `dev-vX.Y-YEAR-MONTH-patches`
+* Same as with the current release, RKE1 changes the default URL PR and points to that new branch
+* Same as with the current release, RKE1 releases an RC and gets tested by QA
+* If QA signs off, that RC becomes the official release
+
+In other words, RKE1 will not wait until Rancher Manager is thorough with its testing to make KDM release official
+
+### Potential problems:
+
+1 - After RKE1 release, `dev-vX.Y-YEAR-MONTH-patches` gets a new PR because rke2 (or other project) requires something else (e.g. bugfix)
+
+RKE1 would have been released using an old `dev-vX.Y-YEAR-MONTH-patches`. However, the difference would be rke2 related code. Therefore, the important RKE1 code in KDM will still be the same. The difference will exist for a short span of time, as it will be removed in the next monthly release `dev-vX.Y-YEAR-MONTH+1-patches` which will include the rke2 bits
```

Comment:
Yeah, RKE1 don't technically care about the the changes in KDM related to rke2, but as of now, we have build logic that checks if the data is the latest or it fails. Could you call this out? We'd need to fix this as a part of changing our process.

Reply:
thanks for this, I was not aware of it. Could you point me to where are we doing that check? Is it automatic?

Reply:
It's in our CI, https://github.com/rancher/rke/blob/267a46d917409dc9f037c3283a079ea48f64af3b/scripts/validate#L19

Reply:
we need to separate rke1 and rke2 in generate logic so data.json changes only if rke1 info changes